src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ioc/InitiatePaymentCallbackHandlerTest.java

```diff
@@ -41,23 +90,20 @@ public void shouldStoreIdAndPaymentNextUrlBeforeSubmittingEvent() throws Excepti
             AboutToStartOrSubmitCallbackResponse.class
         ).getData();
 
-        assertThat(responseData).hasSize(3);
         assertThat(responseData).contains(
-            entry("id", 42),
-            entry("paymentNextUrl", "http://payment_next_url.test"),
-            entry("data", "existingData")
+            entry("channelType", "CITIZEN"),
+            entry("paymentAmount", "1000"),
+            entry("paymentReference", payment.getReference()),
+            entry("paymentStatus", payment.getStatus()),
+            entry("paymentDateCreated", payment.getDateCreated().toLocalDate().toString()),
+            entry("paymentNextUrl", NEXT_URL)
         );
     }
 
     private ResultActions makeRequest(String callbackType) throws Exception {
-        CaseDetails caseDetails = CaseDetails.builder()
-            .id(CASE_ID)
-            .data(ImmutableMap.of("data", "existingData"))
-            .build();
-
         CallbackRequest callbackRequest = CallbackRequest.builder()
             .eventId(CaseEvent.INITIATE_CLAIM_PAYMENT_CITIZEN.getValue())
-            .caseDetails(caseDetails)
+            .caseDetails(successfulCoreCaseDataStoreSubmitResponse())
```

Comment:
We have started using this `successfulCoreCaseDataStoreSubmitResponse()` extensively, which gives single json from a file. How about using `CaseDetails.CaseDetailsBuilder` to build the relevant CaseDetails object instead? We should populate the only fields CaseDetails would expect in specific scenario (i.e. only the fields we pass in initiate payment callback)

Reply:
sure